File: Backjoon/Python/1374_강의실.py
"""
 강의실
 https://www.acmicpc.net/problem/1374
"""

import heapq

n = int(input())
lectures = []
rooms = []
for i in range(n):
    lectures.append(list(map(int,input().split())))
lectures = sorted(lectures, key=lambda x: x[1])
result = 0
for l in lectures:
    if rooms and rooms[0] <= l[1]:
        heapq.heappop(rooms)
    heapq.heappush(rooms, l[2])
    result = max(result, len(rooms))
print(result)

File: Backjoon/1374_강의실.swift
/*
 강의실
 https://www.acmicpc.net/problem/1374
*/

let n = Int(readLine()!)!
var lectures: [[Int]] = []
var rooms: [Int] = []
for i in 0..<n {
    lectures.append(readLine()!.split(separator: " ").map { Int("\($0)")! })
}
lectures.sort { $0[1] < $1[1] }
for l in lectures {
    var isBreak = false
    for i in 0..<rooms.count {
        if rooms[i] <= l[1] {
            rooms[i] = l[2]
            isBreak = true
            break
        }
    }
    if !isBreak {
        rooms.append(l[2])
    }
}
print(rooms.count)
